tests/zkevm/test_worst_compute.py

```diff
@@ -207,35 +206,67 @@ class ReturnDataStyle(Enum):
         ReturnDataStyle.IDENTITY,
     ],
 )
-def test_worst_returndatasize(
+@pytest.mark.parametrize("returned_size", [True, False])
+def test_worst_returndatasize_nonzero(
     state_test: StateTestFiller,
     pre: Alloc,
-    non_zero_size: bool,
+    returned_size: bool,
     return_data_style: ReturnDataStyle,
 ):
     """
-    Test running a block with as many RETURNDATASIZE opcodes as possible.
+    Test running a block which execute as many RETURNDATASIZE opcodes which return a non-zero
+    buffer as possible.
 
-    The `non_zero_size` parameter controls whether opcode must return non-zero value.
+    The `returned_size` parameter indicates the size of the returned data buffer.
     The `return_data_style` indicates how returned data is produced for the opcode caller.
     """
     env = Environment()
 
     dummy_contract_call = Bytecode()
-    if non_zero_size:
-        if return_data_style != ReturnDataStyle.IDENTITY:
-            dummy_contract_call = Op.STATICCALL(
-                address=pre.deploy_contract(
-                    code=Op.REVERT(0, 1)
-                    if return_data_style == ReturnDataStyle.REVERT
-                    else Op.RETURN(0, 1)
-                )
-            )
-        else:
-            dummy_contract_call = Op.MSTORE8(0, 1) + Op.STATICCALL(
-                address=0x04,  # Identity precompile
-                args_size=1,
+    if return_data_style != ReturnDataStyle.IDENTITY:
+        dummy_contract_call = Op.STATICCALL(
+            address=pre.deploy_contract(
+                code=Op.REVERT(0, returned_size)
+                if return_data_style == ReturnDataStyle.REVERT
+                else Op.RETURN(0, returned_size)
             )
+        )
+    else:
+        dummy_contract_call = Op.MSTORE8(0, 1) + Op.STATICCALL(
```

Comment:
Nit: the MSTORE8 is not necessary because if you `args_size=1` and the memory is 0 it will forward the byte of length 1 (with value 0) to the precompile and will thus also return the byte with length 1 (value 0) (thus write that to the return data buffer).

Will nevertheless approve because this is the code prefix and not part of the loop, so should have no significant impact on the cycles.

Reply:
Yeah, I wanted to explicitly put a one because it's easy in the debuggin to see if the test is behaving correctly. Considering MSTORE8 is 3 gas and is out of the loop, I think it is fine.